### ISSUE TITLE
Degraded performance of node's RPC estimatesmartfee after upgrade to  V27.1

### ISSUE THREAD

**Issue by arshad-deriv** [bitcoin/bitcoin#30555]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

I have recently upgraded Bitcoin node from `v26.0` to `v27.1`. I have noticed that RPC calls being `Time Out` specifically for `estimatesmartfee` reaching `120 seconds`. 

### Expected behaviour

If possible, RPC calls like `estimatesmartfee`  should return the response faster.

### Steps to reproduce

Not all the time the mentioned RPC call timeout but there are a few cases after our upgrade to v27.1.

### Relevant log output

There is not much in Bitcoin logs that I analyzed.

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

bitcoin-v27.1-x86_64-linux-gnu

### Operating system and version

linux

### Machine specifications

I am using the  
- Linux OS
- m5.xlarge aws instance
Default config for
- Wallet type is still legacy.
- rpcclienttimeout = 900 sec
- rpcservertimeout = 30 sec
- dbcache = 450
- v2transport = true

**Comment by willcl-ark** [comment]
Yes we will need some logs to debug this further, I see `estimatesmartfee` running extremely quickly on 27.1, so there is likely something else at play here:

<details>
<summary>/usr/bin/time -v bitcoin-cli estimatesmartfee 6 economical</summary>

```
$ /usr/bin/time -v bitcoin-cli estimatesmartfee 6 economical
{
  "feerate": 0.00003446,
  "blocks": 6
}
        Command being timed: "bitcoin-cli estimatesmartfee 6 economical"
        User time (seconds): 0.00
        System time (seconds): 0.00
        Percent of CPU this job got: 100%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:00.00
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 4736
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: 182
        Voluntary context switches: 2
        Involuntary context switches: 0
        Swaps: 0
        File system inputs: 0
        File system outputs: 0
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 0

```

</details>
